docs/proposals/secret-manager-store-api.md

```diff
@@ -0,0 +1,70 @@
+
+## Goals
+The proposed fix for the issue [6671](https://github.com/kubermatic/kubermatic/issues/6671) 
+
+- Define secret using Vault signed certificates to assist with SSH access
+
+## Motivation and Background
+
+A kubermatic user can assign ssh keys to clusters at creation time via the machine controller through cloud-init, but the user ssh agent also assists with updating the keys at runtime via the usersshkeys secret.
+
+The keys are updated mainly via the UserSSHKeys object in the master cluster, so not directly on the secrets. Once the list is updated in the master, users then can attach keys to user clusters which would then create secrets that have the keys in their data field. 
+
+Then those secrets are read from the seed by a user ssh key agent that would always rewrite the ssh keys on each UserSSHKey update.
+
+This approach is great but adds a limitation where the end user has to manually copy credentials to kubermatic usersshkey secrets or added via the API for each cluster's worker nodes authorized_keys.
+
+The new approach documented within this proposal provides the ability to use [SSH CA certificates](https://www.vaultproject.io/docs/secrets/ssh/signed-ssh-certificates), avoiding the burden of copying each user key individually.
+
+This enables authentication via signed keys which can be provided via vault with a lease time configurable by the user.
+
+Vault provides the facility for storing and signing the keys used for authentication.
+
+
+## Implementation
+
+[Signed SSH Certificates](https://www.vaultproject.io/docs/secrets/ssh/signed-ssh-certificates) would be used to implement this.
+
+This includes the following process
+ - Generating a CA key pair 
+ - Configure the SSH daemon to trust the public key via TrustedUserCAKeys configuration, example:
+    ```text
+    .../etc/ssh/sshd_config
+    TrustedUserCAKeys /etc/ssh/trusted-user-ca-keys.pem
+    
+    # Needed for OpenSSH Server 8.1 and older, ssh-rsa is removed from the default supported CA algorithms
+    CASignatureAlgorithms ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521,ssh-ed25519,rsa-sha2-512,rsa-sha2-256,ssh-rsa
+    ```
+ - Starts the SSH daemon with the above configuration on the cluster to use only keys signed by the stated TrustedCA.
+ - Distribute the public key to the workers upon further updates with the help of the `user-ssh-agent`.
+
+> NOTE: This allows the rollout of new CA key pairs for security reasons e.g rotation, revocation etc
+> Management of the CA keys are the responsibility of the admin managing the KKP cluster.
+
+The machine controller would create the CA pem file and configure the SSH daemon to add the `TrustedCAKeys` configuration, the provided CA key is specified on cluster creation.
+
+Distribution of the keys is out of scope from the master to user controller, the secret for the CA would be created on the cluster. The reason for this is to remove the issues around the secret being created after cluster creation.
```

Comment:
The problem to me is that we should be able to provide the public certificate at cluster creation time as the initial machinedeployment should contain it.

> the secret for the CA would be created on the cluster

I'm not sure I understand which cluster you are referring to. If we can skip the distribution of the public keys to start simple it's fine, but take into account that we still need to configure the machines to trust the CA public key so we need it somehow when the md is deployed, isn't it?

This is the controller we use to deployment: https://github.com/kubermatic/kubermatic/blob/9ade0be3abcf1aa63086a533bd232c5d317060ed/pkg/controller/seed-controller-manager/initialmachinedeployment/controller.go#L126

Reply:
@irozzo-1A What about leveraging the user ssh key that we create in the user cluster, by adding a new field which contains the the public key? thus the machine creation wouldn't be dependent anymore on the cloud-init propagation. Once the public is available in the secret, the agent will inject to the machines afterwards in async fashion.

Reply:
This has been done in the implementation

https://github.com/kubermatic/kubermatic/pull/7092